akka-http/src/main/scala/akka/http/scaladsl/coding/Deflate.scala

```diff
@@ -63,10 +63,15 @@ class DeflateCompressor extends Compressor {
     // `flush` and `finish` will return any size depending on the previous input.
     // This value will hopefully provide a good compromise between memory churn and
     // excessive fragmentation of ByteStrings.
-    new Array[Byte](size)
+    // We also make sure that buffer size stays within a reasonable range, to avoid
+    // draining deflator with too small buffer.
+    new Array[Byte](math.max(size, MinBufferSize))
+  }
```

Comment:
Ah, *these* :)
IMO dropping braces is OK for one-liners, but when there's a multi line comment directly above — it makes it a bit confusing.